## PROFITABLE APP PROFILES FOR THE APP STORE AND GOOGLE PLAY MARKETS


In this projext we are going to look at the profiles of the apps that are available in apple store and google play store.For this purpose we are going to access applestore.csv file and googleplaystore.csv file provided by dataquest team. Then we are going to find out what kinds of apps are more getting popular and trace those kinds of apps which have high download ratio.



### Objective/Main Goal

Our goal is to help our developers understand what kinds of apps are likely to attract more users on Google Play and the App Store. To do this, we'll need to collect and analyze data about mobile apps available on Google Play and the App Store.The data is available on googleplaystore.csv and applestore.csv.


#### Opening And Exploring The Data
We are going to open the two data sets we mentioned above, and save both as lists of lists.

The App Store data set is stored in a CSV file named AppleStore.csv, and the Google Play data set is stored in a CSV file named googleplaystore.csv.




In [170]:
from csv import reader
#Apple Store Data
opened_file1=open('AppleStore.csv',encoding='utf8')
read_file1=reader(opened_file1)
ios_data=list(read_file1)
ios_heading=ios_data[0]
ios_apps=ios_data[1:]
#Android Data
opened_file2=open('googleplaystore.csv',encoding='utf8')
read_file2=reader(opened_file2)
playstore_data=list(read_file2)
android_heading=playstore_data[0]
android_apps=playstore_data[1:]
print(len(ios_apps))
print(len(android_apps))
print(android_heading)
print(ios_heading)
print(android_apps[0:2])
print(ios_apps[0:4])
priceofandroid_apps=[]
genres=[]
for items in android_apps:
    price=items[6]
    genre=items[9]
    priceofandroid_apps.append(price)
    genres.append(genre)
print(priceofandroid_apps[0:2])
print(genres[0:2])

7197
10841
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
[['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'], ['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']]
[['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'], ['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+',

To make it easier to explore the two data sets, we will first
write a function named explore_data() so that we can use it
repeatedly to explore rows in a more readable way. We will also
add an option for our function to show the number of rows and
columns for any data set.


In [171]:
def explore_data(dataset,start,end,rows=False,columns=False):
    dataset_slice=dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')    
    if rows and columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        
print(ios_heading)
print('\n')
explore_data(ios_apps,0,5,True,True)
explore_data(android_apps,0,2,True,True)
    
    

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16
['Photo Editor 

After defining explore_data function, we explore the ios_apps data and print the first five rows of ios apps followed by ios headings. We have also found out that there are 7197 ios apps available in appstore and the appstore apps data are classified under 16 different columns of headings.


In [172]:
print(android_heading)
print('\n')
explore_data(android_apps,0,5,True,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

After defining explore_data function, we explore the android_apps data and print the first five rows of android apps followed by android_headings. We have also found out that there are 10841 android apps available in google playstore and the playstore apps data are classified under 13 different columns of headings.


We have to find if there are any missing columns in our ios apps data if we want to clean our data. So,the code below checks whether there is any missing columns in ios apps data.

In [173]:
all_columns=len(ios_heading)
error_in_rows=[]
for items in ios_apps:
    if len(items)!=all_columns:
        error_in_rows.append(items)
print(error_in_rows)


[]


There are not any missing columns in ios apps data so this data doesnot need to delete any rows to make the data clean for analysis and we also need to check for missing columns in android apps data and we can obtain so from the code explained below.

In [174]:
all_columns=len(android_heading)
error_in_rows=[]
error_index=[]
for items in android_apps:
    if len(items)!=all_columns:
        error_in_rows.append(items)
        errorindex=android_apps.index(items)
        error_index.append(errorindex)
print(error_in_rows)
print(error_index)
print(android_heading[8])
for items in error_in_rows:
    print(len(items))
        
        
    

[['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']]
[10472]
Content Rating
12


There is a missing column in android apps in index 10472 of android_apps data which excludes android_heading. The android_heading has 13 columns but at 10472 index of android_data, there are 12 columns and 1 column is missing in this dataset so we have to delete this whole row to make our data clean for analysis.

In [176]:
print(len(android_apps))
print(android_apps[10472])
# del(android_apps[10472])
print(len(android_apps))

10840
['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']
10840


# Finding the apps with duplicate entries and removing duplicate entries



In [177]:
# For android apps 
unique_apps=[]
duplicate_apps=[]
for items in android_apps:
    data=items[0]
    if data in unique_apps:
        duplicate_apps.append(data)
    else:
        unique_apps.append(data)
print(len(android_apps))
print(len(unique_apps))    
print(len(duplicate_apps))
print(duplicate_apps[0:5])

# For ios apps
unique_apps=[]
duplicate_apps=[]
for items in ios_apps:
    data=items[0]
    if data in unique_apps:
        duplicate_apps.append(data)
    else:
        unique_apps.append(data)
print(len(ios_apps))
print(len(unique_apps))    
print(len(duplicate_apps))
print(duplicate_apps[0:5])


10840
9659
1181
['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings']
7197
7197
0
[]


We have written some codes to check whether the apps have duplicated entries or not and we found in our android apps data that there are 1181 apps that have been duplicated but none of the ios apps data have been duplicated. If we analyze without removing duplicate entries from our data, we may not encounter the accuracy of our analysis. So, for improving our accuacy in our analysis, we have to clean the data by removing duplicated entries. However, we cannot randomly delete the duplicated entries. For doing so, we have to look the entries whether all of the data have been duplicated or only few columns are duplicated. If all of the columns of duplicated entries are same then we can remove all duplicated entries otherwise we have to look carefully all the columns and check whether the columns of duplicated entries have been updated after taking another survey. 

# Filtering Duplicate Rows By Using Dictionary


In the code below, we first make an empty dictionary in which we store the android apps name as key and the number of reviews as value. Then we loop through the android_apps data and store the name of apps in a variable inside for loop and we also store the number of reviews in another variable inside for loop. After then we check the condition whether the appname is already stored as a key value or not. If the appname is not stored as a key value in our dictionary then we add the key along with the value associated with the key that is the number of reviews of that app name but if the appname is already stored as a key value in our dictionary then we update the value of the key by the maximum number of reviews of that app and in this way, we prevent the dictionary from duplicate entries by not allowing to add the key name with it's associated value in the dictionary.

In [178]:
reviews_max={}
for items in android_apps:
    name=items[0]
    n_reviews= float(items[3])
    
    if ((name in reviews_max) and (reviews_max[name] < n_reviews)):
        reviews_max[name]=n_reviews
        
    elif name not in reviews_max:
        reviews_max[name]=n_reviews
print(len(reviews_max))

9659


# Filtering Duplicate Entries By Using Lists

At first, we create two empty lists:-One (android_clean) for storing all rows by removing duplicate rows and another(already_added) for storing app names associated with each row by removing all duplicate rows. To do this, we use for loop to iterate through the android_apps data and then we store appname into a variable called 'name' and no of reviews in another variable called 'n_reviews' after typecasting the variable to float inside the for loop. After that we check whether the app is stored in already_added list.If the app is not stored in already_added list the we check whether the numer of reviews stored in our variable 'n_reviews' is equal to the maximum number of reviews for a given app. And finally, we append the whole row in android_clean list and the app name associated with each row in already_added list. 

In [179]:
android_clean=[]
already_added=[]
for data in android_apps:
    name=data[0]
    n_reviews= float(data[3])
   
    
    if ((n_reviews== reviews_max[name]) and (name not in already_added)):
        android_clean.append(data)
        already_added.append(name)
        
print(len(android_clean))
print(len(already_added))



9659
9659


In [180]:
explore_data(android_clean,0,4,True,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows: 9659
Number of columns: 13


In [181]:
print(chr(97))
print(ord('*'))
print(ord('%'))
print(ord('$'))
print(ord('&'))
print(ord('@'))
print(ord('!'))
print(ord('#'))
print(ord('^'))

a
42
37
36
38
64
33
35
94


After removing the rows with duplicate entries, now our aim is to look for those apps whose name starts from English alphabets since we are 

In [182]:
def is_English(string):
    counter=0
    for char in string:
        if(ord(char)>127):
            counter+=1
    if(counter > 3):
        return False
    else:
        return True
        
print(is_English('Instagram'))
print(is_English('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_English('Docs To Go™ Free Office Suite'))
print(is_English('Instachat 😜'))

True
False
True
True


In [183]:
print(ord('™'))
print(ord('😜'))
print(type(android_clean))
print(type(already_added))

8482
128540
<class 'list'>
<class 'list'>


In [184]:
Nonenglish_android_apps=[]
English_android_apps=[]
for data in list(android_clean):
    appname=data[0]
    if(is_English(appname)):
        English_android_apps.append(data)
    elif(is_English(appname)==False):
        Nonenglish_android_apps.append(data)
print(len(Nonenglish_android_apps))
explore_data(Nonenglish_android_apps,0,2,True,True)
explore_data(English_android_apps,0,4,True,True)

45
['Flame - درب عقلك يوميا', 'EDUCATION', '4.6', '56065', '37M', '1,000,000+', 'Free', '0', 'Everyone', 'Education', 'July 26, 2018', '3.3', '4.1 and up']


['သိင်္ Astrology - Min Thein Kha BayDin', 'LIFESTYLE', '4.7', '2225', '15M', '100,000+', 'Free', '0', 'Everyone', 'Lifestyle', 'July 26, 2018', '4.2.1', '4.0.3 and up']


Number of rows: 45
Number of columns: 13
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '96

In [185]:
Nonenglish_ios_apps=[]
English_ios_apps=[]
for items in ios_apps:
    app=items[1]
    if(is_English(app)==True):
        English_ios_apps.append(items)
    elif(is_English(app)==False):
        Nonenglish_ios_apps.append(items)
explore_data(English_ios_apps,0,2,True,True)
explore_data(Nonenglish_ios_apps,0,2,True,True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 6183
Number of columns: 16
['445375097', '爱奇艺PPS -《欢乐颂2》电视剧热播', '224617472', 'USD', '0.0', '14844', '0', '4.0', '0.0', '6.3.3', '17+', 'Entertainment', '38', '5', '3', '1']


['405667771', '聚力视频HD-人民的名义,跨界歌王全网热播', '90725376', 'USD', '0.0', '7446', '8', '4.0', '4.5', '5.0.8', '12+', 'Entertainment', '24', '4', '1', '1']


Number of rows: 1014
Number of columns: 16


In [186]:
Free_android_apps=[]
Paid_android_apps=[]
exception=[]
for items in English_android_apps:
    price=items[6]
    prices=items[7]
    if(prices=='0'):
        Free_android_apps.append(items)
    else:
        Paid_android_apps.append(items)
    if( prices=='0' and price!='Free'):
        exception.append(items)
explore_data(Free_android_apps,0,2,True,True)
explore_data(Paid_android_apps,0,2,True,True)
explore_data(exception,0,1,True,True)
        

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 8864
Number of columns: 13
['TurboScan: scan documents and receipts in PDF', 'BUSINESS', '4.7', '11442', '6.8M', '100,000+', 'Paid', '$4.99', 'Everyone', 'Business', 'March 25, 2018', '1.5.2', '4.0 and up']


['Tiny Scanner Pro: PDF Doc Scan', 'BUSINESS', '4.8', '10295', '39M', '100,000+', 'Paid', '$4.99', 'Everyone', 'Business', 'April 11, 2017', '3.4.6', '3.0 and up']


Number of rows: 750
Number of columns: 13
['Command & Conquer: Rivals', 'FAMILY', 'NaN', '0', 'Varies with device', '0', 'NaN', '0', 'Everyone 10+', 'Strategy', 'June 28, 2018', 'Varies with device', 'Varies with device']




In [187]:
Free_ios_apps=[]
Paid_ios_apps=[]
for items in English_ios_apps:
    price=items[4]
    if(price=='0.0'):
        Free_ios_apps.append(items)
    else:
        Paid_ios_apps.append(items)
explore_data(Free_ios_apps,0,2,True,True)
explore_data(Paid_ios_apps,0,2,True,True)
print(len(Free_ios_apps))
print(Free_ios_apps[0:2])

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 3222
Number of columns: 16
['362949845', 'Fruit Ninja Classic', '104590336', 'USD', '1.99', '698516', '132', '4.5', '4.0', '2.3.9', '4+', 'Games', '38', '5', '13', '1']


['500116670', 'Clear Vision (17+)', '37879808', 'USD', '0.99', '541693', '69225', '4.5', '4.5', '1.1.3', '17+', 'Games', '43', '5', '1', '1']


Number of rows: 2961
Number of columns: 16
3222
[['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'], ['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']]


In [188]:
android_genres={}
for items in Free_android_apps:
    genres=items[9]   
    if(genres not in android_genres):
        android_genres[genres]=1
    elif((genres in android_genres)):
        android_genres[genres]+=1
print(android_genres)
print(len(android_genres))




    

{'Parenting;Music & Video': 6, 'Video Players & Editors;Music & Video': 2, 'Casual;Music & Video': 1, 'Trivia': 37, 'Arcade;Pretend Play': 1, 'Productivity': 345, 'Educational;Creativity': 3, 'Strategy;Action & Adventure': 1, 'Educational;Brain Games': 6, 'House & Home': 73, 'Strategy': 81, 'Sports': 307, 'Finance': 328, 'Casual;Creativity': 6, 'Entertainment': 538, 'Auto & Vehicles': 82, 'Strategy;Education': 1, 'Simulation;Education': 1, 'Weather': 71, 'Education;Brain Games': 3, 'Art & Design;Pretend Play': 1, 'Art & Design;Creativity': 6, 'Dating': 165, 'Events': 63, 'Adventure;Action & Adventure': 3, 'Lifestyle': 345, 'Social': 236, 'Food & Drink': 110, 'Communication;Creativity': 1, 'Educational;Pretend Play': 8, 'Medical': 313, 'Travel & Local': 206, 'Music;Music & Video': 2, 'Strategy;Creativity': 1, 'Role Playing;Brain Games': 1, 'Education;Education': 30, 'Business': 407, 'Casual;Pretend Play': 21, 'Simulation;Action & Adventure': 7, 'Education;Music & Video': 3, 'Casual;Brai

In [189]:
ios_genres={}
for data in Free_ios_apps:
    primary_genre=data[11]   
    if(primary_genre not in ios_genres):
        ios_genres[primary_genre]=1
    elif((primary_genre in ios_genres)):
        ios_genres[primary_genre]+=1
print(ios_genres)
print(len(ios_genres))
print(len(Free_ios_apps))


{'Lifestyle': 51, 'Travel': 40, 'Social Networking': 106, 'Food & Drink': 26, 'Navigation': 6, 'Reference': 18, 'Games': 1874, 'Medical': 6, 'Shopping': 84, 'Utilities': 81, 'Photo & Video': 160, 'Productivity': 56, 'Education': 118, 'Sports': 69, 'Business': 17, 'Health & Fitness': 65, 'Finance': 36, 'Music': 66, 'Entertainment': 254, 'Weather': 28, 'News': 43, 'Catalogs': 4, 'Book': 14}
23
3222


In [190]:
def freq_table(dataset,index):
    table={}
    total=0
    for items in dataset:
        total+=1
        row=items[index]
        if row in table:
            table[row]+=1
        else:
            table[row]=1
            
    table_percentages={}
    for key in table:
        percentage=(table[key]/total)*100
        table_percentages[key]=percentage
    return table_percentages



def display_table(dataset,index):
    tables=freq_table(dataset,index)
    tables_display=[]
    for keys in tables:
        keys_val_as_tuple=(tables[keys],keys)
        tables_display.append(keys_val_as_tuple)
    tables_sorted=sorted(tables_display,reverse=True)
    for entry in tables_sorted:
        print(entry[1], ':', entry[0])
            
    

In [191]:
freq_table(Free_android_apps,9)

{'Action': 3.1024368231046933,
 'Action;Action & Adventure': 0.10153429602888085,
 'Adventure': 0.6768953068592057,
 'Adventure;Action & Adventure': 0.033844765342960284,
 'Adventure;Education': 0.01128158844765343,
 'Arcade': 1.8501805054151623,
 'Arcade;Action & Adventure': 0.12409747292418773,
 'Arcade;Pretend Play': 0.01128158844765343,
 'Art & Design': 0.5979241877256317,
 'Art & Design;Action & Adventure': 0.01128158844765343,
 'Art & Design;Creativity': 0.06768953068592057,
 'Art & Design;Pretend Play': 0.01128158844765343,
 'Auto & Vehicles': 0.9250902527075812,
 'Beauty': 0.5979241877256317,
 'Board': 0.3835740072202166,
 'Board;Action & Adventure': 0.02256317689530686,
 'Board;Brain Games': 0.078971119133574,
 'Books & Reference': 2.1435018050541514,
 'Books & Reference;Education': 0.01128158844765343,
 'Business': 4.591606498194946,
 'Card': 0.45126353790613716,
 'Card;Action & Adventure': 0.01128158844765343,
 'Casino': 0.42870036101083037,
 'Casual': 1.7599277978339352,
 '

In [192]:
display_table(English_android_apps,9)

Tools : 8.602038693571874
Entertainment : 5.793634283336801
Education : 5.231953401289786
Business : 4.358227584772207
Medical : 4.108591637195756
Personalization : 3.900561680882047
Productivity : 3.879758685250676
Lifestyle : 3.775743707093822
Finance : 3.588516746411483
Sports : 3.442895776991887
Communication : 3.2660703141252343
Action : 3.110047846889952
Health & Fitness : 2.995631370917412
Photography : 2.9124193883919283
News & Magazines : 2.600374453921365
Social : 2.485957977948825
Travel & Local : 2.26752652381943
Books & Reference : 2.26752652381943
Shopping : 2.090701060952777
Simulation : 1.9762845849802373
Arcade : 1.9138755980861244
Dating : 1.768254628666528
Casual : 1.7162471395881007
Video Players & Editors : 1.674641148325359
Maps & Navigation : 1.3417932182234242
Puzzle : 1.2377782400665696
Food & Drink : 1.1649677553567712
Role Playing : 1.0817557728312877
Strategy : 0.9777407946744331
Racing : 0.9465363012273768
Libraries & Demo : 0.8737258165175785
Auto & Vehicl

In [193]:
freq_table(English_android_apps,1)

{'ART_AND_DESIGN': 0.6240898689411275,
 'AUTO_AND_VEHICLES': 0.8737258165175785,
 'BEAUTY': 0.5512793842313293,
 'BOOKS_AND_REFERENCE': 2.26752652381943,
 'BUSINESS': 4.358227584772207,
 'COMICS': 0.5720823798627002,
 'COMMUNICATION': 3.2660703141252343,
 'DATING': 1.768254628666528,
 'EDUCATION': 1.1025587684626585,
 'ENTERTAINMENT': 0.9049303099646349,
 'EVENTS': 0.6656958602038693,
 'FAMILY': 19.325982941543582,
 'FINANCE': 3.588516746411483,
 'FOOD_AND_DRINK': 1.1649677553567712,
 'GAME': 9.819013938007073,
 'HEALTH_AND_FITNESS': 2.995631370917412,
 'HOUSE_AND_HOME': 0.7593093405450385,
 'LIBRARIES_AND_DEMO': 0.8737258165175785,
 'LIFESTYLE': 3.786145204909507,
 'MAPS_AND_NAVIGATION': 1.3417932182234242,
 'MEDICAL': 4.108591637195756,
 'NEWS_AND_MAGAZINES': 2.600374453921365,
 'PARENTING': 0.6240898689411275,
 'PERSONALIZATION': 3.900561680882047,
 'PHOTOGRAPHY': 2.9124193883919283,
 'PRODUCTIVITY': 3.879758685250676,
 'SHOPPING': 2.090701060952777,
 'SOCIAL': 2.485957977948825,
 '

In [194]:
display_table(English_android_apps,1)

FAMILY : 19.325982941543582
GAME : 9.819013938007073
TOOLS : 8.61244019138756
BUSINESS : 4.358227584772207
MEDICAL : 4.108591637195756
PERSONALIZATION : 3.900561680882047
PRODUCTIVITY : 3.879758685250676
LIFESTYLE : 3.786145204909507
FINANCE : 3.588516746411483
SPORTS : 3.3804867900977738
COMMUNICATION : 3.2660703141252343
HEALTH_AND_FITNESS : 2.995631370917412
PHOTOGRAPHY : 2.9124193883919283
NEWS_AND_MAGAZINES : 2.600374453921365
SOCIAL : 2.485957977948825
TRAVEL_AND_LOCAL : 2.2779280216351157
BOOKS_AND_REFERENCE : 2.26752652381943
SHOPPING : 2.090701060952777
DATING : 1.768254628666528
VIDEO_PLAYERS : 1.6954441439567296
MAPS_AND_NAVIGATION : 1.3417932182234242
FOOD_AND_DRINK : 1.1649677553567712
EDUCATION : 1.1025587684626585
ENTERTAINMENT : 0.9049303099646349
LIBRARIES_AND_DEMO : 0.8737258165175785
AUTO_AND_VEHICLES : 0.8737258165175785
WEATHER : 0.8217183274391513
HOUSE_AND_HOME : 0.7593093405450385
EVENTS : 0.6656958602038693
PARENTING : 0.6240898689411275
ART_AND_DESIGN : 0.6240

In [195]:
freq_table(English_ios_apps,11)

{'Book': 0.8895358240336406,
 'Business': 0.8571890667960537,
 'Catalogs': 0.08086689309396733,
 'Education': 6.6310852337053205,
 'Entertainment': 7.261846999838266,
 'Finance': 0.7924955523208799,
 'Food & Drink': 0.7116286592269124,
 'Games': 54.860100274947435,
 'Health & Fitness': 2.6686074721009216,
 'Lifestyle': 1.6011644832605532,
 'Medical': 0.3396409509946628,
 'Music': 2.215752870774705,
 'Navigation': 0.452854601326217,
 'News': 0.9218825812712276,
 'Photo & Video': 5.515122109008572,
 'Productivity': 2.7171276079573023,
 'Reference': 0.8571890667960537,
 'Shopping': 1.3747371825974446,
 'Social Networking': 2.037845705967977,
 'Sports': 1.6820313763545207,
 'Travel': 0.9704027171276078,
 'Utilities': 3.4449296458030085,
 'Weather': 1.1159631246967492}

In [196]:
display_table(English_ios_apps,11)

Games : 54.860100274947435
Entertainment : 7.261846999838266
Education : 6.6310852337053205
Photo & Video : 5.515122109008572
Utilities : 3.4449296458030085
Productivity : 2.7171276079573023
Health & Fitness : 2.6686074721009216
Music : 2.215752870774705
Social Networking : 2.037845705967977
Sports : 1.6820313763545207
Lifestyle : 1.6011644832605532
Shopping : 1.3747371825974446
Weather : 1.1159631246967492
Travel : 0.9704027171276078
News : 0.9218825812712276
Book : 0.8895358240336406
Reference : 0.8571890667960537
Business : 0.8571890667960537
Finance : 0.7924955523208799
Food & Drink : 0.7116286592269124
Navigation : 0.452854601326217
Medical : 0.3396409509946628
Catalogs : 0.08086689309396733


In [197]:
app_genre=freq_table(English_ios_apps,11)

for genre in app_genre:
    total=0
    len_genre=0
    for items in English_ios_apps:
        genre_app=items[11]
        if genre_app==genre:
            no_of_users=float(items[5])
            total+=no_of_users
            len_genre+=1
    avg_no_of_user_ratings=total/len_genre

    print(genre, ':' , avg_no_of_user_ratings)

Lifestyle : 8930.373737373737
Travel : 19030.183333333334
News : 16980.315789473683
Social Networking : 60253.84920634921
Food & Drink : 19934.386363636364
Navigation : 19370.821428571428
Games : 15586.759433962265
Medical : 648.952380952381
Utilities : 7927.525821596244
Catalogs : 3465.0
Productivity : 8508.089285714286
Education : 2472.278048780488
Sports : 15350.913461538461
Business : 5149.320754716981
Health & Fitness : 10802.157575757576
Finance : 23353.530612244896
Shopping : 26635.011764705883
Book : 10359.2
Photo & Video : 14688.715542521993
Entertainment : 8862.409799554565
Weather : 23145.246376811596
Reference : 27037.188679245282
Music : 29047.109489051094


In [198]:
app_genre=freq_table(English_ios_apps,11)
ios_apps_ratings={}
ratings_count={}
for items in English_ios_apps:
    genre=items[11]
    noofuserratings=float(items[5])
    if genre not in ios_apps_ratings and genre not in ratings_count:
        ratings_count[genre]=1
        ios_apps_ratings[genre]=noofuserratings
    
    else:
        ratings_count[genre]+=1
        ios_apps_ratings[genre]+=noofuserratings
        
rating_count_by_genre={k: ios_apps_ratings[k]/ratings_count[k] for k in ios_apps_ratings}
print(rating_count_by_genre)
   

    

{'Lifestyle': 8930.373737373737, 'Travel': 19030.183333333334, 'News': 16980.315789473683, 'Social Networking': 60253.84920634921, 'Food & Drink': 19934.386363636364, 'Navigation': 19370.821428571428, 'Games': 15586.759433962265, 'Medical': 648.952380952381, 'Utilities': 7927.525821596244, 'Catalogs': 3465.0, 'Productivity': 8508.089285714286, 'Education': 2472.278048780488, 'Sports': 15350.913461538461, 'Business': 5149.320754716981, 'Health & Fitness': 10802.157575757576, 'Finance': 23353.530612244896, 'Shopping': 26635.011764705883, 'Book': 10359.2, 'Photo & Video': 14688.715542521993, 'Entertainment': 8862.409799554565, 'Weather': 23145.246376811596, 'Reference': 27037.188679245282, 'Music': 29047.109489051094}


In [199]:
ratings_count_by_genre_displaytable=[]
for keys in rating_count_by_genre:
        keys_val_as_tuple=(rating_count_by_genre[keys],keys)
        ratings_count_by_genre_displaytable.append(keys_val_as_tuple)
        ratings_count_by_genre_displaytable_sorted=sorted(ratings_count_by_genre_displaytable,reverse=True)
for entry in ratings_count_by_genre_displaytable_sorted:
    print(entry[1], ':', entry[0])



Social Networking : 60253.84920634921
Music : 29047.109489051094
Reference : 27037.188679245282
Shopping : 26635.011764705883
Finance : 23353.530612244896
Weather : 23145.246376811596
Food & Drink : 19934.386363636364
Navigation : 19370.821428571428
Travel : 19030.183333333334
News : 16980.315789473683
Games : 15586.759433962265
Sports : 15350.913461538461
Photo & Video : 14688.715542521993
Health & Fitness : 10802.157575757576
Book : 10359.2
Lifestyle : 8930.373737373737
Entertainment : 8862.409799554565
Productivity : 8508.089285714286
Utilities : 7927.525821596244
Business : 5149.320754716981
Catalogs : 3465.0
Education : 2472.278048780488
Medical : 648.952380952381


In [200]:
display_table(English_ios_apps,11)

Games : 54.860100274947435
Entertainment : 7.261846999838266
Education : 6.6310852337053205
Photo & Video : 5.515122109008572
Utilities : 3.4449296458030085
Productivity : 2.7171276079573023
Health & Fitness : 2.6686074721009216
Music : 2.215752870774705
Social Networking : 2.037845705967977
Sports : 1.6820313763545207
Lifestyle : 1.6011644832605532
Shopping : 1.3747371825974446
Weather : 1.1159631246967492
Travel : 0.9704027171276078
News : 0.9218825812712276
Book : 0.8895358240336406
Reference : 0.8571890667960537
Business : 0.8571890667960537
Finance : 0.7924955523208799
Food & Drink : 0.7116286592269124
Navigation : 0.452854601326217
Medical : 0.3396409509946628
Catalogs : 0.08086689309396733


In [210]:
cat_freq= freq_table(English_android_apps,1)
for category in cat_freq:
    total = 0
    len_category = 0
    for data in English_android_apps:
        category_app = data[1]
        if category_app == category:
            no_of_installs = data[5]
            no_of_installs1=no_of_installs.replace(',', '')
            no_of_installs2=no_of_installs1.replace('+', '')
            total += float(no_of_installs2)
            len_category += 1
            
    average=total/len_category
    print(category, ':', average)
            
        

SHOPPING : 6966908.880597015
FINANCE : 1319851.4028985507
ART_AND_DESIGN : 1887285.0
GAME : 14256217.600635594
SPORTS : 3373767.6861538463
PRODUCTIVITY : 15530942.008042896
FAMILY : 3345018.516684607
DATING : 828971.2176470588
SOCIAL : 22961790.384937238
AUTO_AND_VEHICLES : 632501.3214285715
COMICS : 817657.2727272727
BUSINESS : 1663758.627684964
WEATHER : 4570892.658227848
TRAVEL_AND_LOCAL : 13218662.767123288
EVENTS : 249580.640625
BEAUTY : 513151.88679245283
LIBRARIES_AND_DEMO : 630903.6904761905
NEWS_AND_MAGAZINES : 9472807.04
LIFESTYLE : 1369954.7774725275
FOOD_AND_DRINK : 1891060.2767857143
BOOKS_AND_REFERENCE : 7641777.871559633
PARENTING : 525351.8333333334
ENTERTAINMENT : 11375402.298850575
EDUCATION : 1782566.0377358492
PHOTOGRAPHY : 16636241.267857144
VIDEO_PLAYERS : 24121489.079754602
TOOLS : 9785955.211352658
HEALTH_AND_FITNESS : 3972300.388888889
HOUSE_AND_HOME : 1331540.5616438356
PERSONALIZATION : 4086652.4853333333
COMMUNICATION : 35153714.17515924
MEDICAL : 96944.4987

In [213]:
category_count={}
category_installs={}
for items in English_android_apps:
    category=items[1]
    no_of_installs=items[5]
    no_of_installs1=no_of_installs.replace(',',' ')
    no_of_installs2=no_of_installs.replace('+',' ')
    if category not in category_count and category not in category_installs:
        category_count[category]=1
        category_installs[category]=float(no_of_installs2)
    else:
        category_count[category]+=1
        category_installs[category]+=float(no_of_installs2)
print(category_count)
print(category_installs)

ValueError: could not convert string to float: '10,000 '